docs/digital_analog_qc/analog-qubo.md

```diff
@@ -98,109 +92,102 @@ Q = np.array(
         [5.42015853, 0.85604541, 0.17675796, 0.32306662, -10.0],
     ]
 )
+
+# Cost function for a single bitstring
+def cost_bitstring(bitstring, qubo_mat):
+    z = np.array(list(bitstring), dtype=int)
+    cost = z.T @ qubo_mat @ z
+    return cost
+
+# Cost function for a full measured sample
+def cost_qubo(counter, qubo_mat):
+    cost = sum(counter[key] * cost_bitstring(key, qubo_mat) for key in counter)
+    return cost / sum(counter.values())  # Divide by total samples
 ```
 
-The analog circuit is composed of two global rotations per layer.  The first
-rotation corresponds to the mixing Hamiltonian and the second one to the
-embedding Hamiltonian in the QAOA algorithm. Subsequently, there is an Ising interaction
-term to emulate the analog circuit. Please note that the Rydberg level is set to 70. We
-initialize the weighted register graph from the QUBO definition similarly to what is
-done in Pulser, and set the device specs with the updated Rydberg level.
+The QAOA algorithm needs a variational quantum circuit with optimizable parameters.
+We use a fully analog circuit composed of two global rotations per layer on
+different axes of the Bloch sphere.
+The first rotation corresponds to the mixing Hamiltonian and the second one to the
+embedding Hamiltonian (given by the register coordinates and equivalent to a
```

Comment:
```suggestion
embedding Hamiltonian (given by the register coordinates and equivalent to a ...)
```

I find this part in parenthesis unclear. Not sure if that relates only to the embedding Hamiltonian or the mixing one or both ?

Reply:
Only the embedding. I will clarify it.